src/thread_event.c

```diff
@@ -71,6 +80,21 @@ thread_prof_sample_event_handler(tsd_t *tsd) {
 	}
 }
 
+static void
+thread_stats_interval_event_handler(tsd_t *tsd) {
+	assert(opt_stats_interval > 0);
+	assert(stats_interval_event_wait_get(tsd) == 0U);
+	uint64_t last_event = thread_allocated_last_event_get(tsd);
+	uint64_t last_stats_event = stats_interval_last_event_get(tsd);
+	stats_interval_last_event_set(tsd, last_event);
+
+	if (stats_interval_accum(tsd, last_event - last_stats_event)) {
+		je_malloc_stats_print(NULL, NULL, opt_stats_interval_opts);
```

Comment:
If the `write_cb` is `NULL`, then it will write to `stderr`. If the printing is automatic, i.e. not user triggered, then it may take place any time, and the application may suddenly get a huge amount of `stderr` content at unpredictable times. Can there be a different way? Perhaps the way `prof_idump` works, i.e. having sequenced output files?

Reply:
Supporting output to file should be useful too. I'd say leave that to a future improvement. On the other hand you can also argue, the stats_interval is for convenience only, if it's intended to be dumping to files etc, users can implement on top of the interval-based user events that we planned.